db/db_impl/db_impl_open.cc

```diff
@@ -1314,6 +1327,62 @@ Status DBImpl::RecoverLogFiles(const std::vector<uint64_t>& wal_numbers,
   return status;
 }
 
+void DBImpl::MoveCorruptedWalFiles(std::vector<uint64_t>& wal_numbers,
+                                   uint64_t corrupted_wal_number) {
+  size_t corrupt_index_start = wal_numbers.size() - 1;
+  size_t i = 0;
+  size_t num_wals = wal_numbers.size();
+  // Find the index of first corrupted wal.
+  while (i < num_wals) {
+    if (wal_numbers[i] == corrupted_wal_number) {
+      corrupt_index_start = i;
+      break;
+    }
+    i++;
+  }
+
+  // Increment i to move WAL files from first corrupted_wal_number + 1.
+  i++;
+
+  std::string archivalPath =
+      ArchivalDirectory(immutable_db_options_.GetWalDir());
+  Status create_status = env_->CreateDirIfMissing(archivalPath);
+
+  // create_status is only checked when it needs to move the corrupted WAL files
+  // to archive folder.
+  create_status.PermitUncheckedError();
+
+  // Move all the WAL files from corrupted_wal_number + 1 to last WAL
+  // (max_wal_number) to avoid column family inconsistency error to archival
+  // directory. If its unable to create archive dir, it will delete the
+  // corrupted WAL files.
+  // We are moving all but first corrupted WAL file to a different folder.
+  while (i < num_wals) {
```

Comment:
iiuc, the unit tests we added so far does not cover the case of (i < num_wals) because the last WAL is corrupted? so there is really no WAL between the first corrupted WAL and the new WAL created during recovery.